testmodule/testContracts.nim

```diff
@@ -114,9 +114,22 @@ suite "Contracts":
     let signer0 = provider.getSigner(accounts[0])
 
     let subscription = await token.subscribe(Transfer, handleTransfer)
-    await token.connect(signer0).mint(accounts[0], 100.u256)
+    discard await token.connect(signer0).mint(accounts[0], 100.u256)
     await subscription.unsubscribe()
 
     await token.connect(signer0).transfer(accounts[1], 50.u256)
 
     check transfers == @[Transfer(receiver: accounts[0], value: 100.u256)]
+
+  test "can wait for contract interaction tx to be mined":
+    # must be spawned so we can get newHeads inside of .wait
+    asyncSpawn provider.mineBlocks(3)
```

Comment:
Instead of spawning, you could explicitly handle the future that is returned by `.wait(3)`:

1. you call `.wait(3)` without `await`, and store the future
2. you call `await provider.mineBlocks(3)`
3. you `await` the future and perform further checks on the receipt

Reply:
Good call -- this gives much better control over block production and makes the tests more stable.